project/code/keychain/store.py

```diff
@@ -0,0 +1,62 @@
+"""
+KeyChain key-value store (stub).
+
+NB: Feel free to extend or modify.
+"""
+
+from keychain import Blockchain
+from keychain import Transaction
+
+
+class Callback:
+    def __init__(self, transaction, chain):
+        self._transaction = transaction
+        self._chain = chain
+
+    def wait(self):
+        """Wait until the transaction appears in the blockchain."""
+        raise NotImplementedError
+
+    def completed(self):
+        """Polls the blockchain to check if the data is available."""
+        raise NotImplementedError
+
+
+class Storage:
+    def __init__(self, bootstrap, miner, difficulty):
+        """Allocate the backend storage of the high level API, i.e.,
+        your blockchain. Depending whether or not the miner flag has
+        been specified, you should allocate the mining process.
+        """
```

Comment:
Shouldn't we instead say that the Storage connects to an (existing and persistent) chain, rather than allocate a new one?

Reply:
> Shouldn't we instead say that the Storage connects to an (existing and persistent) chain, rather than allocate a new one?
That's what the bootstrap address is for :) It will download the peers from the bootstrap node, and start the downloading procedure (by calling /blockchain or something), and continue from there.

Reply:
Alright, maybe specify the purpose of `bootstrap` in the docstring?